probe-rs/src/architecture/arm/core/exception_handling/armv6m_armv7m_shared.rs

```diff
@@ -45,15 +65,130 @@ bitfield! {
     pub apsr_z_bit, _: 30;
     pub apsr_c_bit, _: 29;
     pub apsr_v_bit, _: 28;
+    /// Indicates if the stack was realigned to 8 byte boundary on exception entry.
+    pub stack_was_realigned, _:9;
     pub exception_number, _: 8,0;
 }
 
 /// Decode the exception information.
 pub(crate) fn exception_details(
-    exception_interface: &dyn ExceptionInterface,
+    exception_interface: &impl ExceptionInterface,
     memory_interface: &mut dyn MemoryInterface,
     stackframe_registers: &DebugRegisters,
+    debug_info: &DebugInfo,
 ) -> Result<Option<ExceptionInfo>, Error> {
+    let frame_return_address = get_stack_frame_return_address(stackframe_registers)?;
+
+    if ExcReturn(frame_return_address).is_exception_flag() == 0xF {
```

Comment:
I would prefer an early return in the non matching case instead of indenting the whole if block :)